db/external_sst_file_test.cc

```diff
@@ -314,6 +317,49 @@ TEST_F(ExternalSSTFileTest, ComparatorMismatch) {
   ASSERT_NOK(DeprecatedAddFile({file}));
 }
 
+TEST_F(ExternalSSTFileTest, NoBlockCache) {
+  LRUCacheOptions co;
+  co.capacity = 32 << 20;
+  std::shared_ptr<Cache> cache = NewLRUCache(co);
+  BlockBasedTableOptions table_options;
+  table_options.block_cache = cache;
+  table_options.filter_policy.reset(NewBloomFilterPolicy(10));
+  table_options.cache_index_and_filter_blocks = true;
+  Options options = CurrentOptions();
+  options.table_factory.reset(NewBlockBasedTableFactory(table_options));
+  Reopen(options);
+
+  size_t usage_before_ingestion = cache->GetUsage();
+  std::map<std::string, std::string> true_data;
+  // Ingest with fill_cache = true
+  ASSERT_OK(GenerateAndAddExternalFile(options, {1, 2}, -1, false, false, true,
+                                       false, false, &true_data, nullptr,
+                                       /*fill_cache=*/true));
+  ASSERT_EQ(FilesPerLevel(), "0,0,0,0,0,0,1");
+  EXPECT_GT(cache->GetUsage(), usage_before_ingestion);
+
+  TablePropertiesCollection tp;
+  ASSERT_OK(db_->GetPropertiesOfAllTables(&tp));
+  for (const auto& entry : tp) {
+    EXPECT_GT(entry.second->index_size, 0);
+    EXPECT_GT(entry.second->filter_size, 0);
+  }
+
+  usage_before_ingestion = cache->GetUsage();
+  // Ingest with fill_cache = false
+  ASSERT_OK(GenerateAndAddExternalFile(options, {3, 4}, -1, false, false, true,
+                                       false, false, &true_data, nullptr,
+                                       /*fill_cache=*/false));
+  EXPECT_EQ(usage_before_ingestion, cache->GetUsage());
```

Comment:
Do you know why when this file is written to manifest, this part that loads table handlers into table cache won't load the index and filter blocks into block cache?https://github.com/facebook/rocksdb/blob/2abbb02d144860cffef5b1639a7d2025069cc836/db/version_set.cc#L5545-L5551

Reply:
Yes. It's at line 6031 of db_impl.cc. We pass in a ReadOptions with fill_cache=false to LogAndApply that will be used here to load table handlers.

Reply:
Yeah, this hard codes `prefetch_index_and_filter_in_cache` to be true, and I assume these files don't have a handle in table cache yet, so won't table cache try to create a `TableReader` for it with that `prefix_index_and_filter_in_cache` true setting?

Reply:
> so won't table cache try to create a TableReader for it 

Table cache will create TableReader irrespective of the value of `prefix_index_filter_in_cache`. ReadOptions::fill_cache controls whether we add blocks to block cache when opening this table.

Reply:
We can probably pass in fill_cache for `prefix_index_filter_in_cache`, but leaving it as is should be fine.